utilities/user_utils.py

```diff
@@ -28,9 +29,20 @@ class UserTestSession:
     password: str
     original_user: str
     api_server_url: str
+    is_byoidc: bool = False
 
     def __post_init__(self) -> None:
         """Validate the session data after initialization."""
+        if self.is_byoidc:
+            # we need to grab the username in byoidc mode
+            with self.login():
+                success, user, _ = run_command(command=["oc", "whoami"])
```

Comment:
Does this work, if someone logins in via UI command?

Reply:
I am not sure what you mean here. as long as the shell running the code is logged in with a given context, oc whoami would return the name of the logged in user.